### ISSUE TITLE
mBART tokenizer not following expected target token order

### ISSUE THREAD

**Issue by zijwang** [huggingface/transformers#15957]
## Environment info

- `transformers` version: 4.17
- Platform: linux
- Python version: 3.8.12
- PyTorch version (GPU?): 1.10.2+cu113
- Tensorflow version (GPU?):
- Using GPU in script?:
- Using distributed or parallel set-up in script?:

### Who can help
Maybe  @patil-suraj @SaulLu @patrickvonplaten ?

## Information

Model I am using (Bert, XLNet ...): mBART

The problem arises when using:
* [x] the official example scripts: (give details below)
* [ ] my own modified scripts: (give details below)

The tasks I am working on is:
* [ ] an official GLUE/SQUaD task: (give the name)
* [x] my own task or dataset: (give details below)

## To reproduce & Expected behavior

```
from transformers import MBartTokenizerFast, MBartTokenizer
tokenizer = MBartTokenizer.from_pretrained('facebook/mbart-large-en-ro')
tokenizer.src_lang = 'en_XX'
tokenizer.tgt_lang = 'ro_RO'

print(tokenizer('haha'))
with tokenizer.as_target_tokenizer():
    print(tokenizer('haha'))
```

which returns 
```
{'input_ids': [22010, 2, 250004], 'attention_mask': [1, 1, 1]}
{'input_ids': [22010, 2, 250020], 'attention_mask': [1, 1, 1]}
```

However, per the [official doc](https://huggingface.co/transformers/v4.1.1/model_doc/mbart.html#training), the target sequence should follow the order of `[tgt_lang_code] X [eos]`, which is `[250020, 2, 22010]` in the case above.

**Comment by patil-suraj** [comment]
Hey @zijwang ! Thank you for reporting this.

Here actually the doc is wrong. Here the target tokens actually refer to  `decoder_input_ids` for which the format should be
`[tgt_lang_code] x [eos]`

But it's not required to directly pass `decoder_input_ids`, we can instead just pass `labels` which is of the format `x [eos][tgt_lang_code]`. And the `decoder_input_ids` are prepared inside the model by shifting `labels` to the right, so we get the correct format for `decoder_input_ids` i.e `[tgt_lang_code] x [eos]`.


The `with tokenizer.as_target_tokenizer():` is used to prepare `labels`, so the order is correct.  I will fix the docs and add better examples.

**Comment by zijwang** [comment]
Thanks, @patil-suraj . I see your point here. Having the docs updated will be helpful :)
Meanwhile, do we still need to set `decoder_start_token_id` (something like https://github.com/huggingface/transformers/blob/1a62b25caf06cd4a13af2db1e94abce9969a1d9b/examples/pytorch/translation/run_translation_no_trainer.py#L365) given `decoder_input_ids` is right-shifted input?

**Comment by patil-suraj** [comment]
>  do we still need to set `decoder_start_token_id`

It's not required to set it, if you pass `labels` those will be created inside the model.
